metricsbp/baseplate_hooks.go

```diff
@@ -98,9 +105,24 @@ func (h *spanHook) OnAddCounter(span *tracing.Span, key string, delta float64) e
 	return nil
 }
 
+type countActiveRequestsHook struct {
+	metrics *Statsd
+}
+
+func (h countActiveRequestsHook) OnPostStart(_ *tracing.Span) error {
+	atomic.AddInt64(&h.metrics.activeRequests, 1)
```

Comment:
I still think it's _cleaner_ to have these as methods on the actual object that owns `activeRequests` rather than mutating them separately, this is probably a confusing thing to see from an outside perspective.  They don't have to be exported, but I do think that `Statsd` should _own_ the code that mutates it's internal state.

Reply:
I just feel if we do that, that's would be just a very thin wrap around `atomic.AddInt64` and `atomic.LoadInt64`, which feels really unnecessary.

all the stuff touching it are unexported now (except `RunSysStats`) so there's not really any "outside perspective" 😅

Reply:
Right... what I'm saying is it's icky for one thing to reach into another struct to mutate it's internal state, that seems like a _major_ code smell to me.

Reply:
Fine. Functions added.